packages/flutter/test/material/text_field_test.dart

```diff
@@ -1338,14 +1350,16 @@ void main() {
     );
     // Read only text field cannot open keyboard.
     await tester.showKeyboard(find.byType(TextField));
-    expect(tester.testTextInput.hasAnyClients, false);
+    // On web, we always create a client connection to the engine.
+    expect(tester.testTextInput.hasAnyClients, isBrowser ? isTrue : isFalse);
```

Comment:
I think we should find a way to avoid this distinction. We do this on the web to get the correct context menu, correct? If so, perhaps we can treat "having a client" separately from "being in edit state". The latter is what this test seems to care about. If this makes sense to you, could you please file an issue and leave a TODO with the link?

Reply:
This check is making sure that the framework doesn't open a connection, by mistake, for a read-only text field. It's a valid check for native platforms. But on web, we always need the real input element to perform all the text selection stuff. Few things:

1. Correct context menu (with real copy and select all options).
2. Using the keyboard to manipulate the selection.
3. `cmd/ctrl + c` to copy.